transcom/mymove
# MAIN B-20622 mobile home office api

## [B-20622](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20622)

INT PR: https://github.com/transcom/mymove/pull/13564

## Summary

This is to edit handers and swagger files for ghc related to Mobile Home shipment for office users.

- Added mobile home shipment to MTOShipment ghc API(GET, POST, PATCH).

### How to test
1. Create a move with any shipment.
2. As an office user ->  Go to the move details page for the move you have created.
3. Use the browser dev tools get the info you need for the Swagger or Postman.
4. Use the POST to create a new move and refresh the move details page, and you should see a new mobile home shipment created.
5. Use other api endpoints to test PATCH and GET.

Use Swagger or Postman to check ghc APIs:

Import `ghc.yaml` file from swagger folder if you have not already.

POST - officelocal:3000{{baseUrl}}/mto-shipments
http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/createMTOShipment
Example body to use(use moveTaskOrderID for your move):
```
{
  "moveTaskOrderID": "a71cdc15-c045-4ad6-aa3a-30065cbe4afc",
  "shipmentType": "MOBILE_HOME",
  "mobileHomeShipment": {
    "year": 111,
    "model": "test model",
    "make": "test make",
    "lengthInInches": 122,
    "widthInInches": 822,
    "heightInInches": 71
  }
}
```

PATCH - officelocal:3000{{baseUrl}}/move_task_orders/:moveTaskOrderID/mto_shipments/:shipmentID
http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/updateMTOShipment
Example body to use:
```
{
  "shipmentType": "MOBILE_HOME",
  "boatShipment": {
    "lengthInInches": 133,
    "widthInInches": 111,
    "heightInInches": 123
  }
}
```

GET - http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/listMTOShipments
GET - http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/getShipment

## Screenshots
![image](https://github.com/user-attachments/assets/7b7d3b43-98e4-46b1-a2ce-b2d3d4c5b8d6)
![image](https://github.com/user-attachments/assets/02d509b2-98c6-4e7a-8815-2be55b2717d5)
![image](https://github.com/user-attachments/assets/18af8ee4-32d9-4283-b8c3-bc988840a67d)
![image](https://github.com/user-attachments/assets/afd59576-0aea-4723-a307-c5a8dddc61c4)
